packages/ref-core/src/cmip_ref_core/executor.py

```diff
@@ -18,23 +20,62 @@ class Executor(Protocol):
 
     name: str
 
-    def run_metric(self, metric: Metric, definition: MetricExecutionDefinition) -> MetricResult:
+    def run_metric(
+        self,
+        provider: MetricsProvider,
+        metric: Metric,
+        definition: MetricExecutionDefinition,
+        metric_execution_result: MetricExecutionResult | None = None,
+    ) -> None:
         """
         Execute a metric
 
+        No results are returned from this method,
+        as the execution may be performed asynchrounusly so results may not be immediately available.
+
+        /// admonition | Note
+        In future, we may return a `Future` object that can be used to retrieve the result,
+        but that requires some additional work to implement.
+        ///
```

Comment:
Depending on how we expect we'll run this eventually, it might also make sense to make `run_metric` async. But async python is of course a whole own world that we should only enter if we actually expect to use an event loop in the final daemon.

Reply:
happy to give an overview, maybe in a software meeting or something if it is also interesting for others.